backend/controllers/doctor-controller.js

```diff
@@ -0,0 +1,148 @@
+const { QueryTypes, Sequelize } = require("sequelize");
+const Moment = require("moment");
+const RequiredDetails = require("../models/required-details");
+const Status = require("../models/status");
+const db = require("../config/database");
+
+function getRequiredDetails(req, res) {
+    RequiredDetails.findAll({
+        raw: true,
+        where: {
+            Patient_PatientId: req.params.patientId,
+        },
+    })
+        .then(requiredDetails => {
+            // Renaming the detail names
+            const temp = [];
+            const keys = Object.keys(requiredDetails[0]);
+            for (let i = 1; i < keys.length; i++) {
+                if (keys[i].includes("Required")) {
+                    temp.push({
+                        [keys[i].replace("Required", "")]: requiredDetails[0][keys[i]],
+                    });
+                }
+            }
+            res.json(temp);
+        })
+        .catch(err => {
+            console.log(err);
+            res.status(400).send("Could not find details");
+        });
+}
+
+function updateRequiredDetails(req, res) {
```

Comment:
```
function updateRequiredDetails(req, res) {
    // verify if the patient exists in DB
    const updatedPatient = RequiredDetails.findOne({
        where: { Patient_PatientId: req.body.patientId}
    })
    .then(token => {
        if(token === null) throw new Error("Patient not found in DB")
    })
    .catch(err =>{
        console.log("patient does not exists");
        //console.log("Error: ", err);
        res.status(400).send("Cannot update a non-existing patient !");
    })
    .then(() => {
        RequiredDetails.update(
            {
                WeightRequired: req.body.weight ? "1" : "0",
                TemperatureRequired: req.body.temperature ? "1" : "0",
                SymptomsRequired: req.body.symptoms ? "1" : "0",
            },
            {
                where: { Patient_PatientId: req.body.patientId },
            }
        )
            .then(() => {
                res.status(200).send("Successfully updated details.");
            })
            .catch(err => {
                console.log(err);
                res.status(400).send(`Error: ${err}`);
            });
    })
    
}
```

Try this and let me know if it works. Theoretically, it should not break the update while also integrating the check you want in your test case.

Reply:
I'd test it myself but I somehow can't access the frontend.